### PR TITLE
Feature/campaing and base classes

### PR DESCRIPTION
**Pull request by chelofm**
Added the following:
* Test to Edit Campaign Object
* Test to Delete Campaign Object, for this case was implemented delete action on base class and Details Campaign
* I added a LogManager class
* I added an Object Genie to created Campaign Objects
* I added delete actions on teardown method on TestEditCampaign and TestDeleteCampaign

### CODE REVIEW

File: src/main/java/framework/LogManager.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+/**
+ * Created by Marcelo Ferrufino on 9/1/2015.
+ */
+public class LogManager {
+
+    private static LogManager instance = null;
+    private static Logger logger;
+
+
+    public LogManager (){
```
Comment (svirginia) [inline src/main/java/framework/LogManager.java]: Miss space after ) , please update the other similar places

File: src/main/java/pages/basepages/DetailsBase.java [hunk 2]
```diff
@@ -19,11 +23,41 @@
     @CacheLookup
     protected WebElement editBtn;
 
-    public DetailsBase() {
+    @FindBy(css = "input[value='Delete']")
+    @CacheLookup
+    protected WebElement deleteBtn;
+
+    public DetailsBase(){
```
Comment (svirginia) [inline src/main/java/pages/basepages/DetailsBase.java]: space after)

File: src/main/java/pages/basepages/DetailsBase.java [hunk 3]
```diff
@@ -19,11 +23,41 @@
...
     }
 
     public abstract Object clickEditBtn();
+
+    public abstract Object clickDeleteBtn(boolean confirmDeletion);
+
+    public void clickDeleteButton(boolean confirmDeletion){
+        wait.until(ExpectedConditions.visibilityOf(deleteBtn));
+        deleteBtn.click();
+        Alert alert;
```
Comment (svirginia) [inline src/main/java/pages/basepages/DetailsBase.java]: Then we will need to move the implementation related to alert to webDriverTools class

File: src/main/java/pages/campaign/CampaignDetails.java [hunk 4]
```diff
@@ -76,6 +79,12 @@ public CampaignForm clickEditBtn(){
         return new CampaignForm(driver);
     }
 
+    @Override
+    public CampaignHome clickDeleteBtn(boolean confirmDeletion){
+        clickDeleteButton(confirmDeletion);
```
Comment (svirginia) [inline src/main/java/pages/campaign/CampaignDetails.java]: add super. word if this method is in the parent class

File: src/main/java/pages/campaign/CampaignHome.java [hunk 5]
```diff
@@ -32,4 +32,13 @@ public CampaignDetails selectRecentCpn(String cpnName){
         cpnLnk.click();
         return new CampaignDetails(driver);
     }
+
+    public boolean isCampaignPresent(String cpnName){
+        boolean response = false;
+
+        if ( driver.findElements(By.linkText(cpnName)).size() > 0 ){
```
Comment (svirginia) [inline src/main/java/pages/campaign/CampaignHome.java]: you could return driver.findElements(By.linkText(cpnName)).size() > 0  directly

File: src/main/java/pages/opportunity/OpportunityDetails.java [hunk 6]
```diff
@@ -43,6 +43,11 @@ public Object clickEditBtn() {
         return null;
     }
 
+    @Override
+    public Object clickDeleteBtn(boolean confirmDeletion) {
+        return null;
```
Comment (svirginia) [inline src/main/java/pages/opportunity/OpportunityDetails.java]: Missing to implement this method

File: src/test/java/tests/campaign/CreateCampaign.java [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+    private final String EXPECTED_REVENUE = "100";
+    private final String BUDGETED_COST = "150";
+    private final String ACTUAL_COST = "130";
+    private final String EXPECTED_RESPONSE = "10.00";
+    private final String NUM_SET = "100";
+    private final boolean ACTIVATE = true;
+    private final String TEXT_DESC = "text description";
+
+    @BeforeClass
+    public void setUp(){
```
Comment (svirginia) [inline src/test/java/tests/campaign/CreateCampaign.java]: miss space after )

File: src/main/java/pages/opportunity/OpportunityDetails.java (hunk 6 above)
Comment (chelofm) [inline src/main/java/pages/opportunity/OpportunityDetails.java]: I leave this method to be implemented with Opportunity deletion

File: src/main/java/pages/basepages/DetailsBase.java (hunk 3 above)
Comment (chelofm) [inline src/main/java/pages/basepages/DetailsBase.java]: this method will be moved later.

**Pull request merged**